test/test_datapipe.py

```diff
@@ -497,6 +538,39 @@ def _non_bool_fn(data):
         with self.assertRaises(ValueError):
             temp = list(d for d in filter_dp)
 
+    def test_filter_datapipe_nested_list(self):
+
+        input_ds = IDP(range(10)).batch(5)
+
+        def _filter_fn(data, val):
+            return data >= val
+
+        filter_dp = input_ds.filter(nesting_level=-1, filter_fn=_filter_fn, fn_kwargs={'val': 5})
+        expected_dp = [[5, 6, 7, 8, 9]]
+        self.assertEqual(len(list(zip(filter_dp, expected_dp))), len(expected_dp))
```

Comment:
Can we `assertEqual` between `len(filter_dp)` and `len(expected_dp)`?

Reply:
We cannot call `len(filter_dp)` since `len()` is intentionally not implemented for `filter`